pilot/pkg/bootstrap/server.go

```diff
@@ -882,7 +882,19 @@ func (s *Server) initRegistryEventHandlers() {
 	s.ServiceController().AppendServiceHandler(serviceHandler)
 
 	if s.configController != nil {
-		configHandler := func(_ config.Config, curr config.Config, event model.Event) {
+		configHandler := func(prev config.Config, curr config.Config, event model.Event) {
+			defer func() {
+				if event != model.EventDelete {
+					s.statusReporter.AddInProgressResource(curr)
+				} else {
+					s.statusReporter.DeleteInProgressResource(curr)
```

Comment:
I think this is not right to call this when there is no push

Reply:
https://github.com/istio/istio/pull/34312#discussion_r676164117 Changed it based on that comment.  But can change back if @howardjohn confirms that it is not an issue.